base/show.jl

```diff
@@ -387,7 +387,8 @@ show(io::IO, ::Core.TypeofBottom) = print(io, "Union{}")
 
 function show(io::IO, x::Union)
     print(io, "Union")
-    show_comma_array(io, uniontypes(x), '{', '}')
+    ctx = IOContext(io, :compact => get(io, :compact, false))
+    show_comma_array(ctx, uniontypes(x), '{', '}')
```

Comment:
Shouldn't this be handled directly in `show_comma_array`/`show_delim_array`? What's special about `Union`?

Reply:
Looking through the code more closely, it looks like none of the callers (not even for Array, which computes it itself) are expecting `show_delim_array` to modify the `:compact` parameter.

Aside: this is the only caller of `show_delim_array` (everyone else just specifies the extra `,`) so it seems like we should just do that here too and remove the extra function.

Reply:
Wow, that's right --- the code in arrayshow.jl handles array compactness, so yes it looks to me like `show_delim_array` should just not be touching this at all.